src/business/UserBusiness.ts

```diff
@@ -0,0 +1,101 @@
+import { FriendsDatabase } from "../data/FriendsDatabase"
+import { UserDatabase } from "../data/UserDatabase"
+import { Authenticator } from "../services/Authenticator"
+import { HashManager } from "../services/HashManager"
+import { IdGenerator } from "../services/IdGenerator"
+
+export class UserBusiness {
+
+  public async createUser (name: string, email: string, password: string): Promise<string> {
+    if (!name || !email || !password){
+      throw new Error("Por favor informe todos os dados")
+    }
+
+    if(email.indexOf("@") === -1) {
+        throw new Error("Email inválido")
+    }
+
+    if (password.length < 6) {
+        throw new Error("A senha deve ter no mínimo 6 caracteres")
+    }
+    
+    const idGenerator = new IdGenerator()
+    const newId = idGenerator.generateId()
+
+    const hashManager = new HashManager()
+    const cypherPassword = await hashManager.hash(password)
+
+    const userDatabase = new UserDatabase()
+    await userDatabase.createUser(newId, name, email, cypherPassword)
+
+    const authenticator = new Authenticator()
+    const token = authenticator.generateToken({
+        id: newId
+    })
+
+    return token
+  }
+
+  public async login (email: string, password: string): Promise<string> {
+    if (!email || !password) {
+      throw new Error("Informe email e senha")
+    }
+
+    const userDatabase = new UserDatabase()
+    const user = await userDatabase.getUserByEmail(email)
+
```

Comment:
Seria legal fazer uma validação para ver se o usuário do banco foi encontrado, porque se não for o código quebra

```
if(!user) {
}
```